shared/actions/chat2/index.js

```diff
@@ -665,15 +665,12 @@ const onChatSetConvRetention = (_, action, logger) => {
 const onChatSetConvSettings = (_, action, logger) => {
   const {conv, convID} = action.payload.params
   const conversationIDKey = Types.conversationIDToKey(convID)
-  const newRole =
-    conv &&
-    conv.convSettings &&
-    conv.convSettings.minWriterRoleInfo &&
-    conv.convSettings.minWriterRoleInfo.role
+  const newRole = conv?.convSettings?.minWriterRoleInfo?.role
   const role = newRole && TeamsConstants.teamRoleByEnum[newRole]
+  const cannotWrite = conv?.convSettings?.minWriterRoleInfo?.cannotWrite
   logger.info(`got new minWriterRole ${role || ''} for convID ${conversationIDKey}`)
-  if (role && role !== 'none') {
-    return Chat2Gen.createSaveMinWriterRole({conversationIDKey, role})
+  if (role && role !== 'none' && cannotWrite !== null) {
```

Comment:
just `&& cannotWrite`

Reply:
Even though it can be `false`?

Reply:
oops, right. then I think you should check for `undefined` (according to [here](https://github.com/tc39/proposal-optional-chaining#base-case))